controller/share_manager_controller.go

```diff
@@ -393,11 +393,13 @@ func (c *ShareManagerController) isShareManagerRequiredForVolume(volume *longhor
 	}
 
 	// no active workload, there is no need to keep the share manager around
-	hasActiveWorkload := volume.Status.KubernetesStatus.LastPodRefAt == "" && volume.Status.KubernetesStatus.LastPVCRefAt == "" &&
-		len(volume.Status.KubernetesStatus.WorkloadsStatus) > 0
-	if !hasActiveWorkload {
-		return false
-	}
+	//hasActiveWorkload := volume.Status.KubernetesStatus.LastPodRefAt == "" && volume.Status.KubernetesStatus.LastPVCRefAt == "" &&
+	//	len(volume.Status.KubernetesStatus.WorkloadsStatus) > 0
+	//if !hasActiveWorkload {
+	//	return false
+	//}
+
+	//TODO: counter the csi-attacher AD ticket of this volume, if >= 1, return true
```

Comment:
I am thinking if we need to consider the ticket from users

Reply:
The ticket from user will not need the share manager, right? So we will treated it as a regular ticket

Reply:
Yes. Currently, we won't launch a share manager for a user-attached RWX volume. I am not sure if we need to change the behavior and make the RWX volume attachment logic consistent: As long as an RWX volume is attached with regular mode, Longhorn always launches the share manager pod.

Reply:
I see your point. It sounds reasonable. 

But since this is a refactoring, I think we should not change the behavior when the user attaches it manually. If we want to change it, should we create a new improvement ticket?